67_RayQueryGeometry/main.cpp

```diff
@@ -0,0 +1,1010 @@
+						.stageFlags = asset::IShader::E_SHADER_STAGE::ESS_COMPUTE,
+						.count = 1,
+					},
+					{
+						.binding = 2,
+						.type = asset::IDescriptor::E_TYPE::ET_STORAGE_IMAGE,
+						.createFlags = IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS::ECF_NONE,
+						.stageFlags = asset::IShader::E_SHADER_STAGE::ESS_COMPUTE,
+						.count = 1,
+					}
+				};
+				auto descriptorSetLayout = m_device->createDescriptorSetLayout(bindings);
+
+				const std::array<IGPUDescriptorSetLayout*, ICPUPipelineLayout::DESCRIPTOR_SET_COUNT> dsLayoutPtrs = { descriptorSetLayout.get(), nullptr, nullptr, nullptr};
+				renderPool = m_device->createDescriptorPoolForDSLayouts(IDescriptorPool::ECF_UPDATE_AFTER_BIND_BIT, std::span(dsLayoutPtrs.begin(), dsLayoutPtrs.end()));
+				for (uint32_t i = 0; i < m_maxFramesInFlight; i++)
+					renderDs[i] = renderPool->createDescriptorSet(descriptorSetLayout);
+
+				SPushConstantRange pcRange = { .stageFlags = IShader::E_SHADER_STAGE::ESS_COMPUTE, .offset = 0u, .size = 2 * sizeof(uint64_t) };
+				auto pipelineLayout = m_device->createPipelineLayout({ &pcRange, 1 }, smart_refctd_ptr(descriptorSetLayout), nullptr, nullptr, nullptr);
+
+				IGPUComputePipeline::SCreationParams params = {};
+				params.layout = pipelineLayout.get();
+				params.shader.shader = shader.get();
+				m_device->createComputePipelines(nullptr, { &params, 1 }, &renderPipeline);
+			}
+
+			// write descriptors
+			for (uint32_t i = 0; i < m_maxFramesInFlight; i++)
+			{
+				IGPUDescriptorSet::SDescriptorInfo infos[3];
+				infos[0].desc = gpuTlas;
+				infos[1].desc = smart_refctd_ptr(cameraBuffer);
+				infos[1].info.buffer.size = cameraBuffer->getSize();
+				infos[2].desc = outHDRImageViews[i];
+				infos[2].info.image.imageLayout = IImage::LAYOUT::GENERAL;
+				IGPUDescriptorSet::SWriteDescriptorSet writes[3] = {
+					{.dstSet = renderDs[i].get(), .binding = 0, .arrayElement = 0, .count = 1, .info = &infos[0]},
+					{.dstSet = renderDs[i].get(), .binding = 1, .arrayElement = 0, .count = 1, .info = &infos[1]},
+					{.dstSet = renderDs[i].get(), .binding = 2, .arrayElement = 0, .count = 1, .info = &infos[2]},
+				};
+				m_device->updateDescriptorSets(std::span(writes, 3), {});
+			}
+
+			// camera
+			{
+				core::vectorSIMDf cameraPosition(-5.81655884, 2.58630896, -4.23974705);
+				core::vectorSIMDf cameraTarget(-0.349590302, -0.213266611, 0.317821503);
+				matrix4SIMD projectionMatrix = matrix4SIMD::buildProjectionMatrixPerspectiveFovLH(core::radians(60.0f), float(WIN_W) / WIN_H, 0.1, 10000);
+				camera = Camera(cameraPosition, cameraTarget, projectionMatrix, 1.069f, 0.4f);
+			}
+
+			m_winMgr->show(m_window.get());
+			oracle.reportBeginFrameRecord();
+
+			return true;
+		}
+
+		inline void workLoopBody() override
+		{
+			const auto resourceIx = m_realFrameIx % m_maxFramesInFlight;
+
+			if (m_realFrameIx >= m_maxFramesInFlight)
+			{
+				const ISemaphore::SWaitInfo cbDonePending[] =
+				{
```

Comment:
watch for what @Erfan-Ahmadi is doing on `master` branch and apply a similar change here after merging with `master`